# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: c8a45415dc5b
# Full Hash: c8a45415dc5b0b33b7e9d4b9cd8114ac09a05218
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-05-28 21:48:03
# Description:
#   Bug 1706998: Replace QuotaManager::GetRef().MaybeRecordShutdownStep by static QuotaManager::(Safe)MaybeRecordQuotaClientShutdownStep; r=dom-storage-reviewers,janv
#   
#   There are some few unsafe uses of MaybeRecordShutdownStep where the QuotaManager singleton may be  not (yet or anymore) alive.
#   In order to not add many unnecessary null checks, we drop GetRef and MaybeRecordShutdownStep in favor of
#   QuotaManager::MaybeRecordQuotaClientShutdownStep
# ==============================================================================

diff -r cc21fb9be25a -r c8a45415dc5b dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Fri May 28 05:49:26 2021 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Fri May 28 06:46:04 2021 +0000
@@ -4946,8 +4946,8 @@
 
     mCurrentMaintenance = nullptr;
 
-    QuotaManager::GetRef().MaybeRecordShutdownStep(quota::Client::IDB,
-                                                   "Maintenance finished"_ns);
+    QuotaManager::MaybeRecordQuotaClientShutdownStep(quota::Client::IDB,
+                                                     "Maintenance finished"_ns);
 
     ProcessMaintenanceQueue();
   }
@@ -9808,7 +9808,7 @@
   MOZ_ALWAYS_TRUE(gLiveDatabaseHashtable->Get(Id(), &info));
   MOZ_ALWAYS_TRUE(info->mLiveDatabases.RemoveElement(this));
 
-  QuotaManager::GetRef().MaybeRecordShutdownStep(
+  QuotaManager::MaybeRecordQuotaClientShutdownStep(
       quota::Client::IDB, "Live database entry removed"_ns);
 
   if (info->mLiveDatabases.IsEmpty()) {
@@ -9816,7 +9816,7 @@
                !info->mWaitingFactoryOp->HasBlockedDatabases());
     gLiveDatabaseHashtable->Remove(Id());
 
-    QuotaManager::GetRef().MaybeRecordShutdownStep(
+    QuotaManager::MaybeRecordQuotaClientShutdownStep(
         quota::Client::IDB, "gLiveDatabaseHashtable entry removed"_ns);
   }
 
@@ -15506,12 +15506,9 @@
   gFactoryOps->RemoveElement(this);
 
   // We might get here even after QuotaManagerOpen failed, so we need to check
-  // if we have a quota manager. If we don't, we obviously are not in quota
-  // manager shutdown.
-  if (auto* const quotaManager = QuotaManager::Get()) {
-    quotaManager->MaybeRecordShutdownStep(
-        quota::Client::IDB, "An element was removed from gFactoryOps"_ns);
-  }
+  // if we have a quota manager.
+  quota::QuotaManager::SafeMaybeRecordQuotaClientShutdownStep(
+      quota::Client::IDB, "An element was removed from gFactoryOps"_ns);
 
   // Match the IncreaseBusyCount in AllocPBackgroundIDBFactoryRequestParent().
   DecreaseBusyCount();